### PR TITLE
Remove unnecessary !! and fix fav scene npe

### PR DESCRIPTION
**Pull request by EliKaede**
fix fav scene npe

### CODE REVIEW

**Comment by EliKaede** [comment]
This commit fixes an npe issue on my own phone

File: app/src/main/kotlin/com/hippo/ehviewer/ui/scene/FavoritesScene.kt [hunk 1]
```diff
@@ -150,11 +151,10 @@ class FavoritesScene : SearchBarScene() {
     private val vm: VMStorage by viewModels()
     private var urlBuilder by lazyMut { vm::urlBuilder }
     private var _binding: SceneFavoritesBinding? = null
-    private val binding get() = _binding!!
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/scene/FavoritesScene.kt:153]: Do not inline this.
binding should never be accessed outside of view's lifecycle, so it's fine to use !! here.

File: app/src/main/kotlin/com/hippo/ehviewer/ui/scene/FavoritesScene.kt [hunk 2]
```diff
@@ -150,11 +151,10 @@ class FavoritesScene : SearchBarScene() {
...
     private var mAdapter: GalleryAdapter? = null
-    private val tracker get() = mAdapter!!.tracker!!
     private val showNormalFabsRunnable = Runnable {
-        updateJumpFab() // index: 0, 2
+        val binding = _binding ?: return@Runnable
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/scene/FavoritesScene.kt:156]: Did the NPE you encountered happen here?

File: app/src/main/kotlin/moe/tarsin/coroutines/NamedMutex.kt [hunk 3]
```diff
@@ -3,7 +3,8 @@ package moe.tarsin.coroutines
 import arrow.core.memoize
 import kotlin.contracts.InvocationKind
 import kotlin.contracts.contract
-import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.*
+import moe.tarsin.coroutines.withLock
```
Comment (FooIbar) [inline app/src/main/kotlin/moe/tarsin/coroutines/NamedMutex.kt:7]: What's this?

**Review by FooIbar — CHANGES_REQUESTED** [review]

File: app/src/main/kotlin/com/hippo/ehviewer/ui/scene/FavoritesScene.kt (hunk 1 above)
Comment (EliKaede) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/scene/FavoritesScene.kt:153]: but binding is a member variables, you can use it in every time.

Comment (FooIbar) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/scene/FavoritesScene.kt:153]: You can, but shouldn't.

Comment (EliKaede) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/scene/FavoritesScene.kt:153]: I think binding is a nullable member variables.Should check it's null or not in every time use.

File: app/src/main/kotlin/com/hippo/ehviewer/ui/scene/FavoritesScene.kt (hunk 2 above)
Comment (EliKaede) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/scene/FavoritesScene.kt:156]: Yes,In this Runnable scope, binding should check null or not.

File: app/src/main/kotlin/moe/tarsin/coroutines/NamedMutex.kt (hunk 3 above)
Comment (EliKaede) [reply app/src/main/kotlin/moe/tarsin/coroutines/NamedMutex.kt:7]: sorry about that,it's my editor config make this  change.

File: app/src/main/kotlin/com/hippo/ehviewer/ui/scene/FavoritesScene.kt (hunk 2 above)
Comment (EliKaede) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/scene/FavoritesScene.kt:156]: If you do not want change binding，I can use coroutine to fix this code

Comment (FooIbar) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/scene/FavoritesScene.kt:156]: Yeah cancelling the job on lifecycle destroyed is the correct way.

Comment (EliKaede) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/scene/FavoritesScene.kt:156]: please check #152

**Comment by FooIbar** [comment]
Closed in favor of #152

**Pull request closed without merging**